core/webviz/_markdown.py

```diff
@@ -15,14 +16,61 @@ def __init__(self, md):
         """
         super(Markdown, self).__init__()
         self._md = md
+        math_detected = False
 
-        self._rendered = markdown.markdown(
-            self._md,
+        if '$$' or r'\begin' or r'\end' in self._md:
+            math_detected = True
```

Comment:
Probably good idea to write this as:

```
def contains_math(str):
    return '$$' or r'\begin' or r'\end' in str
...
if contains_math(self._md):
    self.header_elements.add(...
```

Reply:
You could also look at the rendered output:

```
   def contains_math(str):
      return '<p class=...' in str
```
 where the class is whatever the mdx_math inserts.